packages/core/core-flows/src/order/workflows/get-order-detail.ts

```diff
@@ -0,0 +1,159 @@
+import { OrderDTO, OrderDetailDTO } from "@medusajs/types"
+import { deduplicate } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../common"
+
+enum PaymentStatus {
+  NOT_PAID = "not_paid",
+  AWAITING = "awaiting",
+  CAPTURED = "captured",
+  PARTIALLY_CAPTURED = "partially_captured",
+  PARTIALLY_REFUNDED = "partially_refunded",
+  REFUNDED = "refunded",
+  CANCELED = "canceled",
+  REQUIRES_ACTION = "requires_action",
+}
+
+enum FulfillmentStatus {
+  NOT_FULFILLED = "not_fulfilled",
+  PARTIALLY_FULFILLED = "partially_fulfilled",
+  FULFILLED = "fulfilled",
+  PARTIALLY_SHIPPED = "partially_shipped",
+  SHIPPED = "shipped",
+  PARTIALLY_RETURNED = "partially_returned",
+  RETURNED = "returned",
+  CANCELED = "canceled",
+  REQUIRES_ACTION = "requires_action",
+}
+
+export const getOrderDetailWorkflowId = "get-order-detail"
+export const getOrderDetailWorkflow = createWorkflow(
+  getOrderDetailWorkflowId,
+  (
+    input: WorkflowData<{ fields: string[]; order_id: string }>
+  ): WorkflowData<OrderDetailDTO> => {
+    const fields = transform(input, ({ fields }) => {
+      return deduplicate([
+        ...fields,
+        "id",
+        "status",
+        "version",
+        "payment_collections.*",
+        "fulfillments.*",
+      ])
+    })
+
+    const order: OrderDTO = useRemoteQueryStep({
+      entry_point: "orders",
+      fields,
+      variables: { id: input.order_id },
+      list: false,
+      throw_if_key_not_found: true,
+    })
+
+    const aggregatedOrder = transform({ order }, ({ order }) => {
+      const order_ = order as OrderDetailDTO
+
+      const getLastPaymentStatus = (order) => {
```

Comment:
Should we move these outside of the workflow, and write some unit tests for them? I feel like they can get quite complex and easy to get wrong.

Reply:
will move them to utils.